spack/ci-spack-pyprecice-deps-1804.dockerfile

```diff
@@ -1,8 +1,10 @@
 # Build stage with Spack pre-installed and ready to be used
-FROM spack/ubuntu-bionic:latest
-RUN spack env create precice
-# Install dependencies for precice and py-pyprecice
-RUN spack env activate precice \ 
-    && spack add boost@1.74.0 cmake@3.18.4 eigen@3.3.8 openssh petsc@3.14.1 pkgconfig precice@develop py-setuptools py-wheel py-cython py-numpy py-mpi4py \
-    && spack concretize -f \
-    && spack install
+FROM spack/ubuntu-bionic:latest 
+WORKDIR /sources
+# Mount the current sources into the build container
+# and build the default environment
+ADD . /sources
+RUN spack --color=always env create --without-view ci /sources/spack/spack.yaml && \
+    spack --color=always -e ci repo add /sources/spack/repo && \
+    spack --color=always -e ci install --fail-fast --only=dependencies && \
+    spack --color=always clean -a
```

Comment:
I  think we have to evaluate here what is better. The approach have specs in the environment or the new one. Why do you create the environment `ci` using `--without-view ci`?

Reply:
I copied this approach from https://github.com/haampie-spack/ci-example/blob/b98a5e8b1fb7e8aec67707a7eced78efa92b0619/tools/docker/build-env.Dockerfile#L7-L12. So I cannot really give a good reason. I'm not sure, if this is really a satisfactory answer, but I'll try to get through the review with it ;)